src/integrations/prefect-redis/prefect_redis/blocks.py

```diff
@@ -0,0 +1,177 @@
+"""Redis credentials handling"""
+
+from typing import Any, Dict, Optional, Union
+
+import redis
+import redis.asyncio
+from pydantic import Field
+from pydantic.types import SecretStr
+from redis.asyncio.connection import parse_url
+
+from prefect.filesystems import WritableFileSystem
+
+DEFAULT_PORT = 6379
+
+
+class RedisDatabase(WritableFileSystem):
```

Comment:
Updated from `RedisCredentials`. I think `RedisDatabase` better captures the utility of this class.